frame/support/src/storage/mod.rs

```diff
@@ -44,82 +47,108 @@ pub mod types;
 pub mod unhashed;
 pub mod weak_bounded_vec;
 
-#[cfg(all(feature = "std", any(test, debug_assertions)))]
-mod debug_helper {
-	use std::cell::RefCell;
+mod storage_layer_tracker {
+	const STORAGE_LAYER_KEY: &'static [u8] = b":storage_layer:";
+	type Layer = u8;
 
-	thread_local! {
-		static TRANSACTION_LEVEL: RefCell<u32> = RefCell::new(0);
+	pub fn get_storage_layer() -> Layer {
+		crate::storage::unhashed::get_or_default::<Layer>(STORAGE_LAYER_KEY)
```

Comment:
I'm not sure if there could be an optimization that we always return `default` for the top call in the call-stack, since the first call always has `default` we could get rid of the storage access there.

Reply:
how do we know we are in a top call?

In any case, this is not a "real storage read" since it happens in memory always